### DISCUSSION TITLE
config argument: dcRedirectionPolicy.  I want to know some details about this argument.

### DISCUSSION THREAD

**Discussion by Johnite** [uber/cadence#4530] [Q&A]
when I  review configuration, https://cadenceworkflow.io/docs/operation-guide/setup/#static-configuration. there is an argument : dcRedirectionPolicy,  it has a default value, but I want to know some details ,anyone help me?

**Comment by longquanzheng** [comment]
The default value is empty which is same as “noop”. One option available today is “selected-api-forwarding”

https://github.com/uber/cadence/blob/f5ce7cb607d3e4ad8640d40b817b88112f60a9b3/config/development_xdc_cluster0.yaml#L97


When using XDC (global domain) feature to failover a domain from one cluster to another one, client may call the passive cluster to start /signal workflows etc. To have a seamless failover experience, cluster should use this forwarding option to forward those APIs to the active cluster. 

https://github.com/uber/cadence/blob/f5ce7cb607d3e4ad8640d40b817b88112f60a9b3/service/frontend/dcRedirectionPolicy.go#L38

**Comment by longquanzheng** [comment]
Note that “selected-api-forwarding” doesn’t forward worker APIs like “PollForDecisionTask”. This is because forwarding in cross dc is expensive so we recommended to run workers on each cluster.

But in migration mode, to make the experience better, I’m thinking to allow forwarding worker APIs as well.

**Comment by longquanzheng** [comment]
https://github.com/uber/cadence/pull/4531 PR to add documentation and more option to make it easier for cluster migration. Feel free to give suggestion
